cmd/zpool/zpool_main.c

```diff
@@ -5159,10 +5160,10 @@ get_namewidth_iostat(zpool_handle_t *zhp, void *data)
 	    cb->cb_verbose);
 	columns = get_columns();
 
+	if (columns > 42+10 && width > columns - 42)
+		width = columns - 42;
 	if (width < 10)
 		width = 10;
-	if (width > columns - 42)
-		width = columns - 42;
 
```

Comment:
Could you simply this to?:
```C
       width = MAX(10, columns - 42);
```
(totally untested by me)

Reply:
No, that isn't what this does or should do.

Because "width" comes into this code with the desired width as calculated from the device name size and indentation levels in the "status" and "iostat" output.  If the width is between 10 and "columns-42" (assuming columns is 52 or more) it shouldn't be changed.

Reply:
You're right, I see what's going on now.  I just dove into this code and added comments to help me understand how it works.  Would you consider using this (more verbose) patch instead?:
```patch
diff --git a/cmd/zpool/zpool_main.c b/cmd/zpool/zpool_main.c
index d62a6eb..d4c2e4e 100644
--- a/cmd/zpool/zpool_main.c
+++ b/cmd/zpool/zpool_main.c
@@ -5148,23 +5148,44 @@ print_zpool_script_list(char *subcommand)
 /*
  * Set the minimum pool/vdev name column width.  The width must be at least 10,
  * but may be as large as the column width - 42 so it still fits on one line.
+ * NOTE: 42 is the width of the default capacity/operations/bandwidth output
  */
 static int
 get_namewidth_iostat(zpool_handle_t *zhp, void *data)
 {
        iostat_cbdata_t *cb = data;
-       int width, columns;
-
-       width = get_namewidth(zhp, cb->cb_namewidth, cb->cb_name_flags,
+       int ideal_width, columns, leftover_width;
+
+       /* 
+        * Get the pool/vdev name column width we'd ideally want to use.
+        * We say 'ideally' here, because our terminal may not be able to
+        * fit all the characters if it's really narrow.
+        */
+       ideal_width = get_namewidth(zhp, cb->cb_namewidth, cb->cb_name_flags,
            cb->cb_verbose);
+
+       /* Get the number of columns in our terminal */
        columns = get_columns();
 
-       if (width < 10)
-               width = 10;
-       if (width > columns - 42)
-               width = columns - 42;
+       /*
+        * If we remove the space we need for the capacity/operations/bandwidth
+        * columns, calculate how much room would we actually have leftover for
+        * the vdev/pool name.  It could be nothing, or it could be huge amount
+        * if the terminal is wide.
+        *
+        * Note, no matter what we need at least 10 chars.
+        */
+       leftover_width = MIN(10, columns - 42);
 
-       cb->cb_namewidth = width;
+
+       /*
+        * leftover_width may be much larger than what we want if our terminal
+        * is really wide.  In that case, shrink it down to ideal_width.
+        * It may also be smaller than idea_width, but there's nothing we
+        * can do about that, since there's just not enough room in the
+        * terminal.
+        */
+       cb->cb_namewidth = MIN(leftover_width, ideal_width);
 
        return (0);
 }
```

Reply:
Hmm.. I don't think I would describe the algorithm the way you did.

And there is no mention of the "likely" situation where get_columns returns  zero (0) because Linux has no record of what the actual width is.  This bug wasn't triggered by a window that was too narrow (less than 42 characters wide) it was triggered when get_columns returned 0.

It's a petty comment on my part, but I don't see that the MAX() and MIN() calls improve readability in this code.  And the result of MAX(10,columns-42) is not really what's "left over" and so I'm not sure the variables are named clearly either.

How about this pseudo code counter proposal?

```
/* get_namewidth() returns the maximum width of any name in that column for any
 * pool/vdev/device line that will be output.
 */
   width = get_namewidth(...)

/* The width we are calculating is the width of the header and also the padding width
 * for lines that are less than maximum width.  The stats take up 42 characters, so the
 * width available for names is:
 */
  available_width = get_columns() - 42;

/* If the maximum width fits on a screen, then great!  Make everything line up by
 * justifying all lines to the same width.  If that max width is larger than what's available,
 * the name plus stats won't fit on one line, and justifying to that width would cause
 * every line to wrap on the screen.  We only want lines with long names to wrap.
 * Limit the padding to what won't wrap.
 */
   if (width > available_width)  width = available_width ;

/* And regardless of whatever the screen width is (get_columns can return 0 if the
 * width is not known or less than 42 for a narrow terminal) have the width be a
 * minimum of 10.
 */
   if (width < 10)  width = 10;
```

Reply:
That psudocode good, thanks for the added comments.

Reply:
OK.  Let me update the code and the pull request.  See if I can figure out how to do that.

Reply:
OK.  I did a git push to the resolve_10245 branch in my repository.  Do I need to do anything else to make it visible here or update the pull request?  Apologies in advance for the noob question.

Reply:
Force updating your branch is enough, but you just mind address the remaining style nits.  You can run `make checkstyle` locally to check.

Reply:
I don't mind at all.  The style errors were confusing.  Thought I understood what was required.  Thanks for the local test command.